localstack-core/localstack/services/stepfunctions/backend/test_state/execution_worker.py

```diff
@@ -21,11 +22,8 @@ def _get_evaluation_entrypoint(self) -> EvalComponent:
     def _get_evaluation_environment(self) -> Environment:
         return TestStateEnvironment(
             aws_execution_details=self._aws_execution_details,
+            execution_type=StateMachineType.STANDARD,
```

Comment:
Correct, we allow SyncExecutionWorker only for express workflows. In this case however we are instantiating the environment object for TestState api actions, which are evaluated synchronously hence why a TestStateExecutionWorker is a SyncExecutionWorker but that utilises a TestStateEnvironment (and defines a different entry point for the program execution in the for of the first State rather than the first Program)